### PR TITLE
add `doc_values` mapping option to geo_shape field mapping

### PR DESCRIPTION
**Pull request by talevy**
This PR adds support for the `doc_values` field mapping parameter.

`true` and `false` supported by the GeoShapeFieldMapper,
only `false` is supported by the LegacyGeoShapeFieldMapper.

relates #37206

### CODE REVIEW

**Review by imotov — COMMENTED** [review]
I think it might make sense to add tests for everything that we added support for here. I played with `copy_to` for a bit and it looks like it no longer complain about it being present, but it doesn't seem to copy anything.

**Comment by iverase** [comment]
+1 to add tests to the different options.

**Comment by talevy** [comment]
ah, thanks for the reviews. I didn't realize more had to be implemented on that front. thought all the copy_to behavior would be inherited

**Comment by talevy** [comment]
Hi Igor, after re-visiting this / forgetting about it... I realize that it is unclear how to support other options that TypeParsers supports like `copy_to` and such. For this reason, I am only adding `doc_values` parsing support.

File: server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java [hunk 1]
```diff
@@ -695,6 +695,42 @@ public void testPointsOnlyFalseWithTermStrategy() throws Exception {
...
+            .endObject().endObject());
+
+        DocumentMapper defaultMapper = createIndex("test2").mapperService().documentMapperParser()
+            .parse("type1", new CompressedXContent(falseMapping));
+        Mapper fieldMapper = defaultMapper.mappers().getMapper("location");
+        assertThat(fieldMapper, instanceOf(LegacyGeoShapeFieldMapper.class));
+
+        assertFalse(((LegacyGeoShapeFieldMapper)fieldMapper).fieldType().hasDocValues());
+
+        assertFieldWarnings("tree");
```
Comment (imotov) [inline server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java]: Do we test proper defaults and somewhere?

File: server/src/test/java/org/elasticsearch/index/mapper/GeoShapeFieldMapperTests.java [hunk 2]
```diff
@@ -214,6 +214,41 @@ public void testIgnoreMalformedParsing() throws IOException {
...
+
+        DocumentMapper defaultMapper = createIndex("test").mapperService().documentMapperParser()
+            .parse("type1", new CompressedXContent(mapping));
+        Mapper fieldMapper = defaultMapper.mappers().getMapper("location");
+        assertThat(fieldMapper, instanceOf(GeoShapeFieldMapper.class));
+
+        boolean hasDocValues = ((GeoShapeFieldMapper)fieldMapper).fieldType().hasDocValues();
+        assertTrue(hasDocValues);
+
+        // explicit false accept_z_value test
```
Comment (imotov) [inline server/src/test/java/org/elasticsearch/index/mapper/GeoShapeFieldMapperTests.java]: copy/paste error?

**Review by imotov — COMMENTED** [review]
I feel like there are still a few bits missing. For example I would expect it to show up in doXContentBody as well if have some merge handling.

File: server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java (hunk 1 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java]: what do you mean?

Comment (imotov) [reply server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java]: We have different defaults for doc_values for legacy and non-legacy shape. I wonder if we test these defaults.

Comment (talevy) [reply server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java]: added tests for defaults

File: server/src/test/java/org/elasticsearch/index/mapper/GeoShapeFieldMapperTests.java (hunk 2 above)
Comment (talevy) [reply server/src/test/java/org/elasticsearch/index/mapper/GeoShapeFieldMapperTests.java]: fixed!

**Review by iverase — APPROVED** [review]
lgtm

**Pull request merged**